httpx/config.py

```diff
@@ -251,8 +252,13 @@ def __init__(self, http_versions: HTTPVersionTypes = None):
             self.http_versions = {http_versions.upper()}
         elif isinstance(http_versions, HTTPVersionConfig):
             self.http_versions = http_versions.http_versions
+        elif isinstance(http_versions, Iterable):
+            self.http_versions = {
+                version.upper() if isinstance(version, str) else version
+                for version in http_versions
+            }
         else:
-            self.http_versions = {version.upper() for version in http_versions}
+            raise ValueError(f"Unsupported HTTP version {http_versions!r}.")
```

Comment:
I guess we should probably raise a type error instead. Something like...

`TypeError("HTTP version should be a string or list of strings, but got {type(http_versions)}")`?